airflow/models/taskinstance.py

```diff
@@ -3371,6 +3371,10 @@ def tg2(inp):
         ancestor_ti_count = common_ancestor.get_mapped_ti_count(self.run_id, session=session)
         ancestor_map_index = self.map_index * ancestor_ti_count // ti_count
 
+        # The task has not been expanded yet. Let's help it.
+        if self.map_index == -1 and ti_count > 1:
+            return range(0, ancestor_ti_count)
```

Comment:
Hmm, I’d prefer we don’t call this function for this case at all instead (i.e. try to ensure the task is expanded).

Also this check is not correct. If `ti_count` is exactly 1 if the task is mapped (against a list of length 1).